mathosphere-core/src/test/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcherTest.java

```diff
@@ -0,0 +1,86 @@
+package com.formulasearchengine.mathosphere.mathpd.text;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by Moritz on 25.04.2017.
+ */
+public class BasicStringMatcherTest {
+
+    private final Map<Integer, String> testCases;
+    private final BasicStringMatcher matcher = new BasicStringMatcher(2, 4);
+
+    public BasicStringMatcherTest() {
+        testCases = new java.util.HashMap<>();
+        testCases.put(1, "word word word");
+        testCases.put(2, "word   öäüßß\0 word word");
+        testCases.put(3, "word not a word");
+    }
+
+    @org.junit.Test
+    public void compare() throws Exception {
+        Map<Tuple2<Integer, Integer>, Integer> expectedMatchLengths = new HashMap<>();
+        expectedMatchLengths.put(Tuple2.of(1, 1), 1);
+        expectedMatchLengths.put(Tuple2.of(1, 2), 1);
+        expectedMatchLengths.put(Tuple2.of(1, 3), 0);
+        Function<Tuple2<Integer, Integer>, Integer> f = entry -> {
```

Comment:
Kind of - yes. If you declare a functional interface, see: http://stackoverflow.com/questions/27872387/can-a-java-lambda-have-more-than-1-parameter